server/services/store/sqlstore/blocks.go

```diff
@@ -481,3 +481,74 @@ func (s *SQLStore) getBlock(db sq.BaseRunner, c store.Container, blockID string)
 
 	return &blocks[0], nil
 }
+
+func (s *SQLStore) getBlocksWithSameID(db sq.BaseRunner) ([]model.Block, error) {
+	subquery, _, _ := s.getQueryBuilder(db).
+		Select("id").
+		From(s.tablePrefix + "blocks").
+		Having("count(id) > 1").
+		GroupBy("id").
+		ToSql()
+
+	rows, err := s.getQueryBuilder(db).
+		Select(s.blockFields()...).
+		From(s.tablePrefix + "blocks").
+		Where(fmt.Sprintf("id IN (%s)", subquery)).
+		Query()
+	if err != nil {
+		s.logger.Error(`getBlocksWithSameID ERROR`, mlog.Err(err))
+		return nil, err
+	}
+	defer s.CloseRows(rows)
+
+	return s.blocksFromRows(rows)
+}
+
+func (s *SQLStore) replaceBlockID(db sq.BaseRunner, currentID, newID, workspaceID string) error {
```

Comment:
From what I can see, this is only called from the migration code, which should be single-threaded.  Just want to point out that if this gets called concurrently the `runUpdateForBlocksAndHistory` methods will compete and create problems.

Reply:
Right! Data migrations don't lock the database, so multiple nodes will compete to run them. The plugin API has a mechanism to create "advisory locks" using the plugin's `KVStore`, I'll take a stab on using it to sync the nodes and update the PR

Reply:
Addressed in https://github.com/mattermost/focalboard/commit/9cf867ed9e74223083431c407f2d62b91861116a, PTAL